tests/CMakeLists.txt

```diff
@@ -1,248 +1,22 @@
-endforeach()
-
-#   TEST EEG RESULTS ON DIPOLES
-
-# defining variables for those who do not use VTK
-# TODO handle comparison test for NNX1 geom when not USE_VTK...
-
-if (USE_VTK) 
-    set(NNG "NNa" "NNb" "NNc")
-    set(NNGa "NNa")
-    set(NNGb "NNb")
-    set(NNGc "NNc")
-    set(NNGa1 "NNa1")
-    set(NNGb1 "NNb1")
-    set(NNGc1 "NNc1")
-    set(NNGa2 "NNa2")
-    set(NNGb2 "NNb2")
-    set(NNGc2 "NNc2")
-endif()
-
-if (TEST_HEAD3)
-    set(HEAD3 3)
-endif()
-
-set(EPSILON1 0.13)
-set(EPSILON2 0.1)
-set(EPSILON3 0.03)
-foreach(DIP 1 2 3 4 5)
-    foreach(HEADGEO "" ${NNG})
-        foreach(HEADNUM 1 2 ${HEAD3})
-            foreach(COMP mag rdm)
-                set(HEAD "Head${HEADGEO}${HEADNUM}")
-                foreach(ADJOINT "" adjoint adjoint2)
-                    set(BASE_FILE_NAME "${HEAD}-dip.est_eeg${ADJOINT}")
-                    # Compare EEG result with analytical solution obtained with Matlab
-                    OPENMEEG_COMPARISON_TEST("EEG${ADJOINT}EST-dip-${HEAD}-dip${DIP}-${COMP}"
-                        ${BASE_FILE_NAME} analytic/eeg_head${HEADNUM}_analytic.txt -${COMP} -eps ${EPSILON${HEADNUM}} -col ${DIP} -full
-                        DEPENDS EEG${ADJOINT}-dipoles-${HEAD})
-                endforeach()
-            endforeach()
-        endforeach()
-    endforeach()
-endforeach()
-set(EPSILON 0.13)
-if (TEST_HEAD3)
-    foreach(DIP 1 2)
-        foreach(COMP mag rdm)
-            set(BASE_FILE_NAME "Head3-dip-skullscalp.est_eeg")
-            # Compare EEG result with a numerical solution obtained with the tetrahedral FEM
-            OPENMEEG_COMPARISON_TEST("EEGEST-dipSkullScalp-Head3-dip${DIP}-${COMP}"
-                ${BASE_FILE_NAME} analytic/eeg_head3-skullscalp_FEM.txt -${COMP} -eps ${EPSILON} -col ${DIP} -full
-                DEPENDS EEG-dipolesSkullScalp-Head3)
-        endforeach()
-    endforeach()
-endif()
-
-#   Set tests that are expected to fail :
-
-foreach(ADJOINT "" "adjoint" "adjoint2")
-    foreach(HEADGEO ${NNGc1})
-        foreach(DIP 1 2 3 4 5)
-            set_tests_properties(cmp-EEG${ADJOINT}EST-dip-Head${HEADGEO}-dip${DIP}-mag PROPERTIES WILL_FAIL TRUE) # all cmp-EEG-mag NNGc1 tests fail...
-        endforeach()
-    endforeach()
-    foreach(HEADGEO "1" ${NNGa1} ${NNGb1})
-        set_tests_properties(cmp-EEG${ADJOINT}EST-dip-Head${HEADGEO}-dip4-rdm PROPERTIES WILL_FAIL TRUE)
-        set_tests_properties(cmp-EEG${ADJOINT}EST-dip-Head${HEADGEO}-dip5-rdm PROPERTIES WILL_FAIL TRUE)
-    endforeach()
-endforeach()
-
-set(VALIDATION_EIT "${CMAKE_CURRENT_BINARY_DIR}/om_validationEIT")
-if (WIN32)
-    set(VALIDATION_EIT "${EXECUTABLE_OUTPUT_PATH}/om_validationEIT")
+# tests
+if (BUILD_TESTING)
+    OPENMEEG_TEST(check_load_geo
+        load_geo ${OpenMEEG_SOURCE_DIR}/data/Head1/Head1.geom ${OpenMEEG_SOURCE_DIR}/data/Head1/Head1.cond)
+    OPENMEEG_TEST(check_test_mesh_ios
+        test_mesh_ios ${OpenMEEG_SOURCE_DIR}/data/Head1/Head1.tri)
 endif()
```

Comment:
this just moved the tests and it broke appveyor. These tests were passing and now they don't? https://ci.appveyor.com/project/openmeegci/openmeeg/builds/25139666

I don't get the logic here. The diff is huge as you move the majority of the code to a different file.

Reply:
I am starting to understand the problem :)